File: KAKAO/크레인인형뽑기게임.java
//https://school.programmers.co.kr/learn/courses/30/lessons/64061?language=java
import java.util.ArrayList;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        ArrayList<Integer> basket = new ArrayList<Integer>();
        
        for(int i = 0; i < moves.length; ++i){
            for(int j = 0; j < board.length; ++j){
                if (board[j][moves[i]-1] == 0) continue;
                else{
                    basket.add(board[j][moves[i]-1]);
                    board[j][moves[i]-1] = 0;
               
                    if(basket.size() > 1){
                        if(basket.get(basket.size()-1) == basket.get(basket.size()-2)){
                            answer += 2;
                            basket.remove(basket.size()-1);
                            basket.remove(basket.size()-1);
                        }
                    }
                    break;
                }
            }
        }
        return answer;
    }
}

File: KAKAO/크레인인형뽑기게임.kt
class Solution {
    fun solution(board: Array<IntArray>, moves: IntArray): Int {
        var answer = 0
        var basket = mutableListOf<Int>()
        for(move in moves){
            for (row in 0..board.size-1){
				if (board[row][move-1] != 0){
                    basket.add(board[row][move-1])
                    board[row][move-1] = 0
                  	if(basket.size >=2){
                        if(basket[basket.size-1] == basket[basket.size-2]){
                            basket.removeAt(basket.size-1)
                            basket.removeAt(basket.size-1)
                            answer += 2
                        }
                    }
                    break
                }              
            }
        }
        return answer
    }
}